MaxText/layers.py

```diff
@@ -1038,6 +1049,8 @@ def __call__(self,
     inputs = nn.with_logical_constraint(inputs, ('activation_batch', 'activation_length', 'activation_embed'))
 
     # inputs: embedded inputs to the decoder with shape [batch, length, emb_dim]
+    residual = inputs
+    #input_layernorm aka pre_self_attention_layer_norm
```

Comment:
Layernorm is confusing here, we probably need rename it as input_norm. But it can be address in another PR.